lib/rubocop/cop/style/inverse_methods.rb

```diff
@@ -62,6 +65,15 @@ def on_send(node)
           end
         end
 
+        # When comparing classes, `!(Integer < Numeric)` is not the same as
+        # `Integer > Numeric`.
+        def could_be_check_for_class_hierarchy?(lhs, rhs, method)
+          return false unless CLASS_COMPARISON_METHODS.include?(method)
+          return true if lhs.const_type?
```

Comment:
I think this method should check the LHS and RHS constants names.
UpperCamelCase constant is a class or module, and UPPER_SNAKE_CASE constant is not a class or module in almost cases.
So this cop can return `false` in UPPER_SNAKE_CASE cases.

Reply:
I had considered doing this initially. I will update the method to check the name. Do you know if we already have a method that does this or if I will need to write some regex?